nemo/collections/nlp/data/language_modeling/megatron/gpt_sft_dataset.py

```diff
@@ -141,99 +142,120 @@ def __getitem__(self, idx):
                 idx = idx.item()
 
         assert idx < len(self.indexed_dataset)
+        # idx may < 0 because we pad_samples_to_global_batch_size, e.g. id = -1
+        if idx < 0:
+            idx = len(self) + idx
         example = self.indexed_dataset[idx]
         return self._process_example(example)
 
-    def _process_example(self, example):
+    def _process_prompt(self, context: str, label: str, query: str):
```

Comment:
were you going to revert this for this PR?

Reply:
I remove the query part. However, I still leave some refactoring codes to split out `_process_truncation()` and `_process_prompt()`.